src/core/atomic.d

```diff
@@ -294,8 +309,10 @@ in (atomicPtrIsProperlyAligned(here), "Argument `here` is not properly aligned")
 }
 
 /// Ditto
-bool cas(MemoryOrder succ = MemoryOrder.seq,MemoryOrder fail = MemoryOrder.seq,T,V1,V2)(shared(T)* here, V1 ifThis, V2 writeThis) pure nothrow @nogc @trusted
-    if (!is(T == class) && (is(T : V1) || is(shared T : V1)))
+bool cas
+    (MemoryOrder succ = MemoryOrder.seq, MemoryOrder fail = MemoryOrder.seq, T, V1, V2)
+    (shared(T)* here, V1 ifThis, V2 writeThis) pure nothrow @nogc @trusted
+    if (!is(T == class) && (is(V1 : T) || is(V1 : shared T)))
```

Comment:
oh I see there is hidden changes!
what's the logic behind turning the type conversion around?
why is it interesting if `here` is implicitly convertible to `ifThis`? `here` is what's being mutated...

Reply:
You should review commit-by-commit, that will be much easier to see, and you will also see the rationale for it ;)

Reply:
Oh I'm sorry! I somehow looked at the inverse of what you're doing.
This is bizarre. Is it my mistake? Weird bug! O_O

So, I wonder if it's actually necessary for `V1` to participate in the constraint?
It also seems odd for `V1` to participate in the constraint but not `V2`...
I wonder if this `is(V1 : T) || is(V1 : shared T)` could be something like `is(RemoveShared!V1 : T)` ?